Snippets/Snippets_6/Handlers/ReplyingHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Snippets6.Handlers
+{
+    using System.Threading.Tasks;
+    using NServiceBus;
+
+    public class ReplyingHandler : IHandleMessages<RequestDataMessage>
+    {
+        #region ReplyingMessageHandler
+
+        public async Task Handle(RequestDataMessage message, IMessageHandlerContext context)
+        {
+            //Create a response message:
+            var response = new DataResponseMessage
+            {
+                DataId = message.DataId,
+                String = message.String
+            };
+
+            //Underneath the covers, Reply sends a new message to the return address on the message being handled.
+            await context.Reply(response);
```

Comment:
I had one first, the I checked the implementation and realized that Reply is a completely different implementation so I removed the "equivalent".